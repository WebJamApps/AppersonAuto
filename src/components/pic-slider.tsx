import Slider from 'react-slick';
import 'slick-carousel/slick/slick.css';
import 'slick-carousel/slick/slick-theme.css';
import Caption from './PicSlider/caption';

const settings = {
  autoplay: true,
  autoplaySpeed: 3000,
  infinite: true,
  speed: 1000,
  slidesToShow: 1,
  slidesToScroll: 1,
  arrows: false,
  fade: true,
};
export interface IpicData {
  _id: string, url: string, title: string, comments: string
}
export function PicSlider({ data }: { data: IpicData[] }): JSX.Element {
  return (
    <div>
      <Slider {...settings}>
        {
          Array.isArray(data) ? data.map((d) => (
            <div key={d._id}>
              {' '}
              <img className="slide-images" src={d.url} alt={d.title} />
              {' '}
              {d.comments === 'showCaption' ? <Caption caption={d.title} /> : null}
            </div>
          ))
            : null
        }
      </Slider>
    </div>
  );
}

