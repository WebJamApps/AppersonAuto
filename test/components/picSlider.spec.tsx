import renderer from 'react-test-renderer';
import { PicSlider } from 'src/components/pic-slider';

describe('picture slider component test', () => {
  it('renders the component', () => {
    const data:any = [
      { _id: 1, url: '../static/imgs/ohaf/slideshow2.png', comments: 'showCaption' },
      { _id: 2, url: '../static/imgs/ohaf/slideshow3.png' },
      { _id: 3, url: '../static/imgs/ohaf/slideshow4.png' },
      { _id: 4, url: '../static/imgs/ohaf/slideshow5.png' },
      { _id: 5, url: '../static/imgs/ohaf/slideshow6.png' },
    ];
    const result = renderer.create(<PicSlider data={data} />).toJSON();
    expect(result).toMatchSnapshot();
  });
  it('renders null when no array of data', () => {
    const data = [{} as any];
    const result:any = renderer.create(<PicSlider data={data} />).toJSON();
    expect(result.type).toBe('div');
  });
});
