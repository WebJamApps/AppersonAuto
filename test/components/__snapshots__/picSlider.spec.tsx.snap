// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`picture slider component test renders the component 1`] = `
<div>
  <div
    className="slick-slider slick-initialized"
    dir="ltr"
  >
    <div
      className="slick-list"
      onClick={[Function]}
      onKeyDown={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={null}
      onMouseMove={null}
      onMouseUp={[Function]}
      onTouchCancel={null}
      onTouchEnd={[Function]}
      onTouchMove={null}
      onTouchStart={[Function]}
      style={Object {}}
    >
      <div
        className="slick-track"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onMouseOver={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <div
          aria-hidden={false}
          className="slick-slide slick-active slick-current"
          data-index={0}
          onClick={[Function]}
          style={
            Object {
              "left": -0,
              "opacity": 1,
              "outline": "none",
              "position": "relative",
              "transition": "opacity 1000ms ease, visibility 1000ms ease",
              "width": 0,
            }
          }
          tabIndex="-1"
        >
          <div>
            <div
              style={
                Object {
                  "display": "inline-block",
                  "width": "100%",
                }
              }
              tabIndex={-1}
            >
               
              <img
                className="slide-images"
                src="../static/imgs/ohaf/slideshow2.png"
              />
               
              <div
                className="slider-caption"
                style={
                  Object {
                    "backgroundColor": "#fff",
                    "boxShadow": "0 4px 5px 0 rgba(0, 0, 0, 0.14), 0 1px 10px 0 rgba(0, 0, 0, 0.12), 0 2px 4px -1px rgba(0, 0, 0, 0.14)",
                    "fontWeight": 600,
                    "marginBottom": "10px",
                    "padding": "15px 0",
                    "textAlign": "center",
                  }
                }
              >
                 
                
                 
              </div>
            </div>
          </div>
        </div>
        <div
          aria-hidden={true}
          className="slick-slide"
          data-index={1}
          onClick={[Function]}
          style={
            Object {
              "left": -0,
              "opacity": 0,
              "outline": "none",
              "position": "relative",
              "transition": "opacity 1000ms ease, visibility 1000ms ease",
              "width": 0,
            }
          }
          tabIndex="-1"
        >
          <div>
            <div
              style={
                Object {
                  "display": "inline-block",
                  "width": "100%",
                }
              }
              tabIndex={-1}
            >
               
              <img
                className="slide-images"
                src="../static/imgs/ohaf/slideshow3.png"
              />
               
            </div>
          </div>
        </div>
        <div
          aria-hidden={true}
          className="slick-slide"
          data-index={2}
          onClick={[Function]}
          style={
            Object {
              "left": -0,
              "opacity": 0,
              "outline": "none",
              "position": "relative",
              "transition": "opacity 1000ms ease, visibility 1000ms ease",
              "width": 0,
            }
          }
          tabIndex="-1"
        >
          <div>
            <div
              style={
                Object {
                  "display": "inline-block",
                  "width": "100%",
                }
              }
              tabIndex={-1}
            >
               
              <img
                className="slide-images"
                src="../static/imgs/ohaf/slideshow4.png"
              />
               
            </div>
          </div>
        </div>
        <div
          aria-hidden={true}
          className="slick-slide"
          data-index={3}
          onClick={[Function]}
          style={
            Object {
              "left": -0,
              "opacity": 0,
              "outline": "none",
              "position": "relative",
              "transition": "opacity 1000ms ease, visibility 1000ms ease",
              "width": 0,
            }
          }
          tabIndex="-1"
        >
          <div>
            <div
              style={
                Object {
                  "display": "inline-block",
                  "width": "100%",
                }
              }
              tabIndex={-1}
            >
               
              <img
                className="slide-images"
                src="../static/imgs/ohaf/slideshow5.png"
              />
               
            </div>
          </div>
        </div>
        <div
          aria-hidden={true}
          className="slick-slide"
          data-index={4}
          onClick={[Function]}
          style={
            Object {
              "left": -0,
              "opacity": 0,
              "outline": "none",
              "position": "relative",
              "transition": "opacity 1000ms ease, visibility 1000ms ease",
              "width": 0,
            }
          }
          tabIndex="-1"
        >
          <div>
            <div
              style={
                Object {
                  "display": "inline-block",
                  "width": "100%",
                }
              }
              tabIndex={-1}
            >
               
              <img
                className="slide-images"
                src="../static/imgs/ohaf/slideshow6.png"
              />
               
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
